promql/analyzer.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promql
+
+import (
+	"errors"
+	"time"
+
+	"golang.org/x/net/context"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+
+	"github.com/prometheus/prometheus/storage/local"
+)
+
+// An Analyzer traverses an expression and determines which data has to be requested
+// from the storage. It is bound to a context that allows cancellation and timing out.
+type Analyzer struct {
+	ng *Engine // The engine that launched the Analyzer.
```

Comment:
Hmm, the PR description lists `analyzer.go` as "no changes", but I guess there are some?